logstash-core/src/main/java/org/logstash/plugins/AliasRegistry.java

```diff
@@ -100,13 +101,51 @@ private YamlWithChecksum(final String yamlContents, final String checksumHash) {
 
         @SuppressWarnings("unchecked")
         private Map<PluginType, List<AliasPlugin>> decodeYaml() throws IOException {
-            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-            return mapper.readValue(yamlContents, new TypeReference<Map<PluginType, List<AliasPlugin>>>() {});
+            Yaml yaml = new Yaml();
+            Map<String, Object> yamlMap = yaml.load(yamlContents);
+
+            // Convert the loaded YAML content to the expected type structure
+            Map<PluginType, List<AliasPlugin>> result = convertYamlMapToObject(yamlMap);
+
+            return result;
```

Comment:
Inlining of the variable.

```suggestion
            return convertYamlMapToObject(yamlMap);
```